ext/http/00_serve.js

```diff
@@ -653,33 +653,46 @@ async function serve(arg1, arg2) {
 
   onListen({ port: listenOpts.port });
 
-  while (true) {
-    const rid = context.serverRid;
-    let req;
-    try {
-      req = await op_http_wait(rid);
-    } catch (error) {
-      if (ObjectPrototypeIsPrototypeOf(BadResourcePrototype, error)) {
+  const serverPromise = new Deferred();
+
+  const server = {
+    finished: serverPromise.promise,
+  };
+
+  // Run the server
+  (async () => {
```

Comment:
Could we make this async IIFE the finished promise itself?

Reply:
Done